File: LCP.py
#!/usr/bin/python
    def longestCommonPrefix(self, strs):
        """
        :type strs: List[str]
        :rtype: str
        """
        if(len(strs) > 1):
                strs.sort(key = lambda x:len(x))
                short = strs[0]
                for i in range(len(short)):
                    for every in strs:
                        if every[i] != short[i]:
                            if(i == 0):
                                return ""
                            return str(short[:i])
                return short
        elif(len(strs) == 1):
            return strs[0]
        else:
            return ""

File: LCP.cpp
string longestCommonPrefix(vector<string>& strs) {
        if(strs.size() == 0)
            return "";
        string lcp = "";
        sort(strs.begin(),strs.end(),[](string& a,string& b){return a.length() < b.length();});
        for(int i = 0;i < strs[0].length();i++)
        {
            for(int j = 1;j < strs.size();j++)
            {
                if(strs[j][i] != strs[0][i])
                    return lcp;
                
            }
            lcp += strs[0][i];
        }
         return lcp;   
    }
